src/radiation/radiation.cpp

```diff
@@ -0,0 +1,621 @@
+    for (int i = 1; i < 20; i++) {
+        this->RayProduct(0, i);
+    }*/
+
+}
+
+/// Declaring function for the initialization to call, draws each ray vector and gets all of the cells associated with it (sorted by distance and starting at the boundary working in)
+void ablate::radiation::RadiationSolver::RayInit() {
+    //ablate::solver::CellSolver::Setup();
+
+    double theta;  // represents the actual current angle (inclination)
+    double phi;    // represents the actual current angle (rotation)
+
+    /// Locally get a range of cells that are included in this subdomain at this time step for the ray initialization
+    // March over each cell in this region to create the stencil
+    // IS cellIS;
+    // PetscInt cStart, cEnd;  // Keep these as local variables, the cell indices in the radiation initializer should be updated with every radiation initialization. The variables here are owned by the
+                            // DM
+    stencilSet.clear(); //Clear the existing set because a new set of cells will be created below
+
+    solver::Range cellRange;
+    GetCellRange(cellRange);
+    // std::set<PetscInt> stencilSet; //Moved to a class variable
+    for (PetscInt c = cellRange.start; c < cellRange.end; ++c) {
+        // if there is a cell array, use it, otherwise it is just c
+        const PetscInt cell = cellRange.points ? cellRange.points[c] : c;
+
+        // keep a list of cells in the stencil
+        stencilSet.insert(cell);
+    }
+
+    /// Create a nested vector which can store cell locations based on origin cell, theta, phi, and space step
+    std::vector<PetscInt> rayPhis;                                // Preallocate the sub-vectors in order to avoid dynamic sizing as much as possible
+    std::vector<std::vector<PetscInt>> rayThetas(nPhi, rayPhis);  // Preallocate the sub-vectors in order to avoid dynamic sizing as much as possible
+    std::vector<std::vector<std::vector<PetscInt>>> rayCells(
+        nTheta, rayThetas);                    // Make vector to store this dimensional row //Preallocate the sub-vectors in order to avoid dynamic sizing as much as possible
+    rays.resize(stencilSet.size(), rayCells);  // Indices: Cell, angle (theta), angle(phi), space steps
+
+    /// Get setup things for the position vector of the current cell index
+    const PetscScalar* cellGeomArray;  // Declare the variables that will contain the geometry of the cells
+    PetscReal minCellRadius;
+    DM cellDM;
+    VecGetDM(cellGeomVec, &cellDM);
+    DMPlexGetGeometryFVM(cellDM, nullptr, &cellGeomVec, &minCellRadius);  // Obtain the geometric information about the cells in the DM?
+    VecGetArrayRead(cellGeomVec, &cellGeomArray);
+    PetscFVCellGeom* cellGeom;
+
+    PetscInt ncells = 0;
+    for (auto iCell : stencilSet) {
+        // double percentComplete = 100.0 * double(ncells) / double(stencilSet.size());
+        // PetscPrintf(PETSC_COMM_WORLD, "Radiation Initializer Percent Complete: %f\n", percentComplete);
```

Comment:
I think that we should use a [ablate::monitors::logs::Log](https://ablate.dev/content/simulations/components/ablate::monitors::logs::Log.html) for progress variable.  That way by default there is not log so nothing reports, but we could provide a log if we want to report the status.